### PR TITLE
New post: Can I run existing Go applications on AWS Lambda? (Part 2)

### PR DESCRIPTION
**Pull request by abhirockzz**
# Content Pull Request Checklist

Please check to make sure your content submission meets the following requirements by adding an `x` between the brackets:

- [x] The title of this pull request reflects the title of my content.
- [x] I have reviewed the [Content Review Checklist](https://github.com/build-on-aws/content/blob/main/CONTENT_REVIEW_CHECKLIST.md).
- [x] This pull request contains one piece of content (post or page).

Is this pull request linked to a Content Proposal or Idea Suggestion Issue? If so, please link to it below using the number or full link.

```text
Resolves #5

or

Content for https://github.com/build-on-aws/content/issues/5
```

**By submitting this pull request, I confirm I own this content, have permission to use each image included in this content, and that it will be released under the [CC BY-SA 4.0 SA License](/LICENSE).**

### CODE REVIEW

**Comment by scribwell** [comment]
@cobusbernard This blog post is ready for your one-step review.

**Comment by bketelsen** [comment]
this needs to be relocated into the tutorials folder like part 1 @abhirockzz

File: posts/golang-gin-app-on-aws-lambda/index.md [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+- In the `Handler` implementation: 
+  - The `Proxy` (or `ProxyWithContext`) method of the `ginadapter.GinLambda` object receives the `events.APIGatewayProxyRequest`, converts it into a `http.Request` object, and sends it to the `gin.Engine` for routing.
+  -  It returns a proxy response object (`events.APIGatewayProxyResponse`) generated from the data written to the response writer (`http.ResponseWriter`).
+
+## Application overview
+
+The sample application presented in this blog is a trimmed down version of `bit.ly` or other solutions you may have used. It uses [Amazon DynamoDB](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Introduction.html?sc_channel=el&sc_campaign=datamlwave&sc_content=golang-gin-app-on-aws-lambda&sc_geo=mult&sc_country=mult&sc_outcome=acq) for persistence and exposes `HTTP` endpoints to access the application (basic CRUD operations).
+
+Although we will not be discussing the application code in detail, it's important to understand the basic structure. Here is the package structure for the URL shortener application (in the [function](https://github.com/build-on-aws/golang-gin-app-on-aws-lambda/tree/main/function) directory):
```
Comment (jennapederson) [inline posts/golang-gin-app-on-aws-lambda/index.md:67]: Some of these github links 404 as code does not yet exist (even in the currently private repo).

File: posts/golang-gin-app-on-aws-lambda/index.md [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+It's time to deploy the URL shortener application and give it a go!
+
+## Prerequisites
+
+Before you proceed, make sure you have the [Go programming language](https://go.dev/dl/) (**v1.18** or higher) and [AWS CDK](https://docs.aws.amazon.com/cdk/v2/guide/getting_started.html#getting_started_install?sc_channel=el&sc_campaign=datamlwave&sc_content=golang-gin-app-on-aws-lambda&sc_geo=mult&sc_country=mult&sc_outcome=acq) installed.
+
+Clone the GitHub repository and change to the right directory:
+
+```bash
+git clone TODO
```
Comment (jennapederson) [inline posts/golang-gin-app-on-aws-lambda/index.md:92]: These TODOs need to be completed.

File: posts/golang-gin-app-on-aws-lambda/index.md [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+  - aws-cdk
+authorGithubAlias: abhirockzz
+authorName: Abhishek Gupta
+date: 2023-07-21
+---
+
+|ToC|
+|---|
+
+[The first part of this series](https://community.aws/posts/golang-gin-app-on-aws-lambda) introduced you to the [AWS Lambda Go API Proxy](https://github.com/awslabs/aws-lambda-go-api-proxy), and how it's framework/package specific adapter implementations (for `gorilla/mux`, `echo` and `net/http`) allows you to run existing Go applications as [AWS Lambda](https://docs.aws.amazon.com/lambda/latest/dg/welcome.html?sc_channel=el&sc_campaign=datamlwave&sc_content=golang-gin-app-on-aws-lambda&sc_geo=mult&sc_country=mult&sc_outcome=acq) functions fronted by [Amazon API Gateway](https://docs.aws.amazon.com/apigateway/latest/developerguide/welcome.html?sc_channel=el&sc_campaign=datamlwave&sc_content=golang-gin-app-on-aws-lambda&sc_geo=mult&sc_country=mult&sc_outcome=acq). If you haven't already, I encourage you to [take a look at it](https://community.aws/posts/golang-gin-app-on-aws-lambda) in order to get a basic understanding of the AWS Lambda Go API Proxy.
```
Comment (jennapederson) [inline posts/golang-gin-app-on-aws-lambda/index.md:18]: Links to content on this site should not use the hardcoded domain. See [here](https://github.com/build-on-aws/content/blob/main/FAQ.md#how-do-i-link-to-other-posts-on-the-communityaws-site) for examples.

Also, these will need to be updated with the path to the series folder once that is published.

**Review by jennapederson — CHANGES_REQUESTED** [review]
Waiting on github repo to be completed and for first post in series to be published.

**Review by cobusbernard — APPROVED** [review]

**Pull request merged**